pkg/kn/traffic/compute_test.go

```diff
@@ -51,169 +53,188 @@ func newTestTrafficCommand() (*cobra.Command, *flags.Traffic) {
 			[]string{"echo-v1", "@latest"},
 			[]string{"latest", ""},
 			[]int{98, 2},
 		},
-		// set 2% to @latest with tag (append it in traffic block)
 		{
+			"set 2% to @latest with tag (append it in traffic block)",
 			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("latest", "echo-v1", 100, false)),
 			[]string{"--traffic", "@latest=2,echo-v1=98", "--tag", "@latest=testing"},
 			[]string{"echo-v1", "@latest"},
 			[]string{"latest", "testing"},
 			[]int{98, 2},
 		},
-		// change traffic percent of an existing revision in traffic block, add new revision with traffic share
 		{
+			"change traffic percent of an existing revision in traffic block, add new revision with traffic share",
 			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("v1", "echo-v1", 100, false)),
 			[]string{"--tag", "echo-v2=v2", "--traffic", "v1=10,v2=90"},
 			[]string{"echo-v1", "echo-v2"},
 			[]string{"v1", "v2"},
 			[]int{10, 90}, //default value,
 		},
-		// append new non-@latest target
-		{
-			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("v1", "echo-v1", 100, false)),
-			[]string{"--tag", "echo-v2=v2", "--traffic", "v1=10,v2=90"},
-			[]string{"echo-v1", "echo-v2"},
-			[]string{"v1", "v2"},
-			[]int{10, 90}, //default value,
-		},
-		// untag
 		{
+			"untag 'latest' tag from 'echo-v1' revision",
 			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("latest", "echo-v1", 100, false)),
 			[]string{"--untag", "latest"},
 			[]string{"echo-v1"},
 			[]string{""},
 			[]int{100},
 		},
+		{
+			"untag 'latest' tag from '@latst' revision",
+			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("latest", "", 100, true)),
+			[]string{"--untag", "latest"},
+			[]string{"@latest"},
+			[]string{""},
+			[]int{100},
+		},
+		{
+			"replace revision pointing to 'latest' tag from 'echo-v1' to 'echo-v2' revision",
+			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("latest", "echo-v1", 50, false), newTarget("", "echo-v2", 50, false)),
+			[]string{"--untag", "latest", "--tag", "echo-v1=old,echo-v2=latest"},
+			[]string{"echo-v1", "echo-v2"},
+			[]string{"old", "latest"},
+			[]int{50, 50},
+		},
 	} {
-		testCmd, tFlags := newTestTrafficCommand()
-		testCmd.SetArgs(testCase.inputFlags)
-		testCmd.Execute()
-		err, targets := Compute(testCmd, testCase.existingTraffic, tFlags)
-		if err != nil {
-			t.Fatal(err)
-		}
-		for i, target := range targets {
-			if testCase.desiredRevisions[i] == "@latest" {
-				assert.Equal(t, *target.LatestRevision, true)
-			} else {
-				assert.Equal(t, target.RevisionName, testCase.desiredRevisions[i])
+		t.Run(testCase.name, func(t *testing.T) {
+			if lper, lrev, ltag := len(testCase.desiredPercents), len(testCase.desiredRevisions), len(testCase.desiredTags); lper != lrev || lper != ltag {
+				t.Fatalf("length of desird revisions, tags and percents is mismatched: got=(desiredPercents, desiredRevisions, desiredTags)=(%d, %d, %d)",
```

Comment:
*desired

Reply:
We don't need this, this would raise dev-time list index out of range error which dev should fix.

And as u have added the test case names, that should be easier to check, I'd recommend removing the len check altogether.

WDYT ?

Reply:
Yeah, it may easy to find due to the test case names, but IMO it scared me and made me suspect that we have a potential bug in our core implementations (not test code). This  simple length check shows me the route cause of this index-out-of-range error at a glance.

Reply:
okay, TBH I should've added code comments in tests about this requirement.